# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/RangeUtils.h
# Commit: a86c7713f936
# Full Hash: a86c7713f936f7e692d673d596dfcf527faccb6d
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-05-10 09:29:34
# Regressor Bug: 1932150
# File Overlap Count: 2
# Description:
#   Bug 1932150 - Fix some inconsistency about indicating whether the selection is allowed to cross the shadow boundary r=jjaschke,dom-core
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D246949
# ==============================================================================

diff -r d0d0f8f22432 -r a86c7713f936 dom/base/RangeUtils.h
--- a/dom/base/RangeUtils.h	Fri May 09 17:09:50 2025 +0000
+++ b/dom/base/RangeUtils.h	Fri May 09 17:16:36 2025 +0000
@@ -25,23 +25,29 @@
 struct ShadowDOMSelectionHelpers {
   ShadowDOMSelectionHelpers() = delete;
 
-  static nsINode* GetStartContainer(const AbstractRange* aRange,
-                                    bool aAllowCrossShadowBoundary);
+  static nsINode* GetStartContainer(
+      const AbstractRange* aRange,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 
-  static uint32_t StartOffset(const AbstractRange* aRange,
-                              bool aAllowCrossShadowBoundary);
+  static uint32_t StartOffset(
+      const AbstractRange* aRange,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 
-  static nsINode* GetEndContainer(const AbstractRange* aRange,
-                                  bool aAllowCrossShadowBoundary);
+  static nsINode* GetEndContainer(
+      const AbstractRange* aRange,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 
-  static uint32_t EndOffset(const AbstractRange* aRange,
-                            bool aAllowCrossShadowBoundary);
+  static uint32_t EndOffset(
+      const AbstractRange* aRange,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 
-  static nsINode* GetParentNodeInSameSelection(nsINode& aNode,
-                                               bool aAllowCrossShadowBoundary);
+  static nsINode* GetParentNodeInSameSelection(
+      const nsINode& aNode,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 
-  static ShadowRoot* GetShadowRoot(const nsINode* aNode,
-                                   bool aAllowCrossShadowBoundary);
+  static ShadowRoot* GetShadowRoot(
+      const nsINode* aNode,
+      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
 };
 }  // namespace dom
 